lambdas/common/deployment_utils.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+# -*- encoding: utf-8 -*-
+"""
+Shared library to help surface ECS deployment information.
+"""
+
+import collections
+import datetime
+
+from ecs_utils import get_cluster_arns, get_service_arns, describe_service
+
+
+Deployment = collections.namedtuple(
+    'Deployment',
+    'deployment_key deployment_status color created_at task_definition'
+)
+DeploymentKey = collections.namedtuple('DeploymentKey', 'id service_arn')
+
+
+DATE_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+def _create_deployment_tuple_from_item(item):
+    item_date = datetime.datetime.strptime(item['created_at'], DATE_FORMAT)
+
+    return Deployment(
+        DeploymentKey(item['deployment_id'], item['service_arn']),
+        item['deployment_status'],
+        item['color'],
+        item_date,
+        item['task_definition']
+    )
+
+
+def _create_deployment_tuple_from_ecs(service, deployment):
+    """Takes am AWS ECS API Service & Deployment, returns a namedtuple."""
+    deployment_status = deployment['status']
+    ongoing_deployment = len(service['deployments']) > 1
+
+    color = "blue"
+    if ongoing_deployment:
+        color = "green" if(deployment_status == "PRIMARY") else "blue"
```

Comment:
style nit: space after `if`

And rather than mixing an inline `if` and an indented `if`, I'd roll it up as follows:

```python
if ongoing_deployment and (deployment_status == 'PRIMARY'):
    color = 'green'
else:
    color = 'blue'
```